ckanext/geocat/utils/xpath_utils.py

```diff
@@ -224,28 +264,33 @@ def xpath_get_distribution_from_distribution_node(
                 description_node)
     else:
         distribution['description'] = {'en': '', 'it': '', 'de': '', 'fr': ''}
-    normed_protocol, protocol_name = _get_normed_protocol(protocol)
-    distribution['protocol'] = normed_protocol
-    distribution['protocol_name'] = protocol_name
+
+    normed_protocol = _get_normed_protocol(protocol)
+    distribution['protocol'] = protocol
+    distribution['normed_protocol'] = normed_protocol
+
     if normed_protocol == DOWNLOAD_PROTOCOL and protocol.startswith(DOWNLOAD_PROTOCOL + ':'):  # noqa
         format = protocol.replace(DOWNLOAD_PROTOCOL + ':', '')
-        media_type = protocol.replace(DOWNLOAD_PROTOCOL + ':', '')
-        distribution['format'] = format
-        distribution['media_type'] = media_type
-    resource_formats = filter(
-        lambda i: i not in [LINKED_DATA_PROTOCOL, MAP_PROTOCOL],
-        SERVICE_PROTOCOLS)
-    if normed_protocol in resource_formats:
+        if format:
+            distribution['media_type'] = format
+            distribution['format'] = format
```

Comment:
so madia_type and format are almost the same at the end?

Reply:
yes the way we handle them is at stake anyway with the apdaption to DCAT-AP: in DCAT-AP format is a controlled vocabulary while media_type is not. See here for how we currently handle the format: https://github.com/opendata-swiss/ckanext-switzerland-ng/blob/a8ae080f9fee370938d71e7240bb8e92486a61e5/ckanext/switzerland/helpers/format_utils.py#L54

I think the correct way here would be to only set the media_type for download resources and not the format or to also consult the format.yaml and do the exact thing that we do interactively in ckaext-switzerland-ng. 

For now maybe it would be best to fill just the media_type. If you want we can discuss this.